node_modules/gh-core/lib/db.js

```diff
@@ -253,6 +253,23 @@ var _setUpModel = function(sequelize) {
             'isGlobalAdmin': function() {
                 return false;
             },
+            'hide': function(ctx) {
+                // There's no need to hide information if the current user is requesting their
+                // own profile or the current user is can administrate the user's application
+                if (ctx.user && (ctx.user.id === this.id || ctx.user.canAdmin(this.AppId))) {
+                    return;
+                }
+
+                // Otherwise we need to hide some information
+                delete this.dataValues.authenticationStrategy;
```

Comment:
It's probably ok, it's not like we're going to add lots of User preferences and if we do, the logic to hide them is right there. That said, I'll write a test to ensure only the properties we expect are returned.